puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/infra/updateJobs.groovy

```diff
@@ -0,0 +1,53 @@
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup workspace') {
+            steps {
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: 'master']],
+                    extensions: [[$class: 'CleanCheckout']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/foreman-infra.git']
+                    ]
+                ])
+
+                dir("./ci") {
+                    sh """
+                    virtualenv venv
+                    source venv/bin/activate
+                    pip install jenkins-job-builder
+                    deactivate
+                    """
+                }
+            }
+        }
+
+        stage('Update ci.centos.org jobs') {
+            steps {
+                dir("./ci/centos.org") {
+                    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'jenkins-centos', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME']]) {
+                        sh """
+                        source ../venv/bin/activate
+                        jenkins-jobs --conf ./centos_jenkins.ini --user ${env.USERNAME} --password ${env.PASSWORD} update --delete-old -r ./jobs
+                        deactivate
+                        """
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        always {
+            script {
+                if(fileExists('ci/venv')) {
```

Comment:
do we need this safeguard?

Reply:
It's probably not strictly needed as if it errored out and the file isn't there it won't matter if it fails here. But I figured it wouldn't hurt to do it a "better" way.

Reply:
`dir('blah')` creates `blah` if it does not exist, so it won't error :)